src/typescript/azure-openapi-validator/rules/DescriptionNoIdenticalSiblings.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+import * as jp from 'jsonpath';
+import { trimDescription } from './utilities';
+import { JsonPath } from '../../jsonrpc/types';
+
+export const DescriptionNoIdenticalSiblings: string = "DescriptionNoIdenticalSiblings";
+const minimumDescriptionLength = 20;
+
+var checkedPaths: string[] = [];
+
+rules.push({
+  id: "D4003",
+  name: DescriptionNoIdenticalSiblings,
+  severity: "error",
+  category: "DocViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.doc,
+  appliesTo_JsonQuery: "$..*[?(@.description)]",
```

Comment:
Maybe we can modify this query to pick all the parent nodes to begin with instead of going to each `description` node and then trying to validate its parent node

Reply:
As stated in the code itself:

```
// JSONPath does not have a parent operator, so we need to do some manipulation
// of the 'path', along with an additional search (and keep a record of the parents)
// in order to compare multiple nodes at once.
```

The primary issue is that JSONPath will not accept `@.*.description` as a query, when this is the exact term required here - we care about any of the current node's children having a description.

Reply:
D'oh! did not know that!

Reply:
It totally didn't make sense to me either, and sucks. Especially since other JSON search specs do include the ability to query on available children/get a parent.

Reply:
I hear ya